### PR TITLE
[infra] Add GLib2.0 cmake config

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit adds GLib2.0 cmake config.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft PR: #9267

### CODE REVIEW

**Comment by chunseoklee** [comment]
How about using the same verison of tizen's glib ? 

https://review.tizen.org/gerrit/gitweb?p=platform/upstream/glib.git;a=blob;f=packaging/glib2.spec;h=023fe78bdda23d940c4d8cc0a057f6b7d672dd6a;hb=15e379116716504dc5865badde2fdcd5e799fcc7 (maybe 2.62?)

**Comment by jyoungyun** [comment]
We are using the glibc version(2.27) that is installed by default on Ubuntu Bionic.
https://launchpad.net/ubuntu/bionic/+source/glibc

```
$ ldd --version
ldd (Ubuntu GLIBC 2.27-3ubuntu1.5) 2.27
```

And I checked the package installed in Tizen rootfs for a while, it installed glibc v2.30 version. The related tizen platform code is [here](https://review.tizen.org/gerrit/gitweb?p=platform/upstream/glib2.0.git;a=blob;f=packaging/glib2.spec;h=2ed8717c659519d1121ba739f5b764488a80714a;hb=refs/heads/tizen).

```
12:59:43 [   58s] dlog-config-persistent-logs-7.0.4-130.########################################
12:59:43 [   58s] glibc-2.30-15.10                      warning: /etc/ld.so.conf saved as /etc/ld.so.conf.rpmorig
12:59:43 [   58s] ########################################
12:59:43 [   58s] libgcc-9.2.0-16.16                    
```

Your link is pointed to glib, not glib2.0 so it doesn't seem to match the installed version.

Should we keep the two versions to be same while forcibly updating the default version installed by Ubuntu?

**Comment by chunseoklee** [comment]
> The related tizen platform code is [here](https://review.tizen.org/gerrit/gitweb?p=platform/upstream/glib2.0.git;a=blob;f=packaging/glib2.spec;h=2ed8717c659519d1121ba739f5b764488a80714a;hb=refs/heads/tizen).

This repo haven't updated after 2013. Is this repo is the one tizen 7.0 image currently used? 

Anyway, we have to find a way to support at least both arm Ubuntu bionic and tizen 7.0. I will also investigate more on this. I am not 100% understand the status.

**Comment by hseok-oh** [comment]
Which package is installed on tizen if we use `BuildRequires: pkgconfig(glib-2.0)` ?

**Comment by hseok-oh** [comment]
>How about using the same verison of tizen's glib ?

Do we need exact version?

**Comment by jyoungyun** [comment]
Oh, it's my mistake.
The current tizen system installed the glib v2.70 version.
```
12:59:44 [   59s] libglib-2.70.0-36.1                   ########################################
```
And this package is from tizen branch of [glib.git](https://review.tizen.org/gerrit/gitweb?p=platform/upstream/glib.git;a=summary).
https://review.tizen.org/gerrit/gitweb?p=platform/upstream/glib.git;a=blob;f=packaging/glib2.spec;h=d070e13b967c162d6cc842645bd4d5e19623e1f2;hb=refs/heads/tizen

**Comment by jyoungyun** [comment]
The Ubuntu bionic installed the glib2.0 v2.56 by default.
https://launchpad.net/ubuntu/bionic/+source/glib2.0
```
$ dpkg-query -l *glib2.0*
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  libglib2.0-0:a 2.56.4-0ubun amd64        GLib library of C routines
```

In tizen build, it will use the glib v2.70, and in ubuntu x64, it will use the glib v2.56 installed by default from Ubuntu Bionic. Should we support the same version of glib everywhere?

**Comment by chunseoklee** [comment]
> >How about using the same verison of tizen's glib ?
> 
> Do we need exact version?

IDK. Let's figure it out :) Since two versions differ not much, we may not need to match two versions.

**Comment by chunseoklee** [comment]
IMHO, it seems to work on any platform if using non-deprecated API.

File: infra/nnfw/cmake/packages/GLib2.0Config.cmake [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  find_library(GLIB_LIBRARIES
+    NAMES glib-2.0
+    ONLY_CMAKE_FIND_ROOT_PATH)
+  
+  get_filename_component(GLIB_LIBRARY_DIR ${GLIB_LIBRARIES} DIRECTORY)
+  find_path(GLIBCONFIG_INCLUDE_DIR
+    NAMES glibconfig.h
+    PATHS ${GLIB_LIBRARY_DIR}
+    PATH_SUFFIXES glib-2.0/include
+    CMAKE_FIND_ROOT_PATH_BOTH)
```
Comment (hseok-oh) [inline infra/nnfw/cmake/packages/GLib2.0Config.cmake:11]: Why this `find_path` use `CMAKE_FIND_ROOT_PATH_BOTH`?

File: infra/nnfw/cmake/packages/GLib2.0Config.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  find_path(GLIB_INCLUDE_DIR
+    NAMES glib.h
+    PATH_SUFFIXES glib-2.0
+    ONLY_CMAKE_FIND_ROOT_PATH)
```
Comment (hseok-oh) [inline infra/nnfw/cmake/packages/GLib2.0Config.cmake:16]: By toolchain file, `find_library` and `find_path` are check rooted directory only. Any reason to use `ONLY_CMAKE_FIND_ROOT_PATH` explicitly?

File: infra/nnfw/cmake/packages/GLib2.0Config.cmake (hunk 1 above)
Comment (jyoungyun) [reply infra/nnfw/cmake/packages/GLib2.0Config.cmake:11]: It considers the case where the `glibconfig.h` file is not located in `GLIB_LIBRARY_DIR`.

Comment (hseok-oh) [reply infra/nnfw/cmake/packages/GLib2.0Config.cmake:11]: My comment means why we need to remain possibility to search host by setting `CMAKE_FIND_ROOT_PATH_BOTH`.

Comment (hseok-oh) [reply infra/nnfw/cmake/packages/GLib2.0Config.cmake:11]: `glibconfig.h` is under `lib/` directory, so we pass `GLIB_LIBRARY_DIR` as hint. `GLIB_LIBRARY_DIR` is host path, so we need to find `glibconfig.h` based on host. 
It's better to use `NO_CMAKE_FIND_ROOT_PATH`, and add comment about this.

File: infra/nnfw/cmake/packages/GLib2.0Config.cmake (hunk 2 above)
Comment (hseok-oh) [reply infra/nnfw/cmake/packages/GLib2.0Config.cmake:16]: IMO, this `find_path` and above `find_library` don't need `ONLY_CMAKE_FIND_ROOT_PATH`.

File: infra/nnfw/cmake/packages/GLib2.0Config.cmake (hunk 1 above)
Comment (jyoungyun) [reply infra/nnfw/cmake/packages/GLib2.0Config.cmake]: Thank you. I applied your suggestion to the latest patch.

**Comment by jyoungyun** [comment]
PTAL /cc @hseok-oh @chunseoklee

**Review by hseok-oh — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**